### DISCUSSION TITLE
add percentage option to mapim

### DISCUSSION THREAD

**Discussion by indus** [libvips/libvips#2231] [Ideas]
I think it would be a nice addition to mapim to make  a percentage option available. Then a single index with float values (between 0 and 1) would map a bunch of inputs of varying size to always the same output.

**Comment by jcupitt** [comment]
Sorry I didn't reply. I agree, this would be a nice (and simple) thing to add.

You can do it now, of course, eg.:

```python
relative_index_image = ....
index_image = pyvips.Image.xyz(relative_index_image.width, relative_index_image.height) * relative_index_image
out = in.mapim(index_image)
```

but there should be a small speedup from avoiding the call to `xyz`.

**Comment by indus** [comment]
No problem. I'm basically doing what you've suggested now. I didn't wanted to bother you longer with an issue labeled 'bug'...Am 28.04.2021 17:44 schrieb John Cupitt ***@***.***>:
Sorry I didn't reply. I agree, this would be a nice (and simple) thing to add.
You can do it now, of course, eg.:
relative_index_image = ....
index_image = pyvips.Image.xyz(in.width, in.height) * relative_index_image
out = in.mapim(index_image)
but there should be a small speedup from avoiding the call to xyz.

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.
Am 28.04.2021 17:44 schrieb John Cupitt ***@***.***>:
Sorry I didn't reply. I agree, this would be a nice (and simple) thing to add.
You can do it now, of course, eg.:
relative_index_image = ....
index_image = pyvips.Image.xyz(in.width, in.height) * relative_index_image
out = in.mapim(index_image)
but there should be a small speedup from avoiding the call to xyz.

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.
Am 28.04.2021 17:44 schrieb John Cupitt ***@***.***>:
Sorry I didn't reply. I agree, this would be a nice (and simple) thing to add.
You can do it now, of course, eg.:
relative_index_image = ....
index_image = pyvips.Image.xyz(in.width, in.height) * relative_index_image
out = in.mapim(index_image)
but there should be a small speedup from avoiding the call to xyz.

**Comment by jcupitt** [comment]
Ah gotcha. I've moved this to "discussions" and tagged it as an idea.
